packages/web/src/features/translation/TranslationToolbar.tsx

```diff
@@ -47,10 +50,38 @@ export function TranslationToolbar({
   };
 
   const userCan = useAccessControl();
+  const isTranslator = userCan('translate', {
+    type: 'Language',
+    id: languageCode,
+  });
+
+  const navigateToNextUnapprovedVerse = useCallback(async () => {
+    const data = await apiClient.verses.findNextUnapprovedVerse(
+      verseId,
+      languageCode
+    );
+    if (data && data.verseId) {
+      onVerseChange(data.verseId);
+    } else {
+      // TODO: figure out how to handle the situation where ALL words have been glossed (i.e. data.verseId === undefined)
+      flash.error('No unapproved verses');
+    }
+  }, [onVerseChange, flash, verseId, languageCode]);
```

Comment:
Should we be using just verseId rather than data.verseId in here? Regardless the verseId in the dependency array needs to be the same as the verseId used in the function or the reactive stuff won't work right.

Reply:
My naming was confusing. `verseId` refers to the current verse. `data.verseId` is the next unapproved verse id (`data` here is of type `GetNextUnapprovedVerse`, which, on a side note, should be named `GetNextUnapprovedVerseResponseBody`). I'll change the names be more clear and consistent :+1:

Reply:
Done.

Reply:
Oh my bad. I misunderstood how this was working.